pkg/rehearse/rehearse_test.go

```diff
@@ -405,3 +405,91 @@ func TestFilterJobsByRequested(t *testing.T) {
 		})
 	}
 }
+
+func TestCreatePairs(t *testing.T) {
+	testCases := []struct {
+		name           string
+		expected       []string
+		expectedErr    bool
+		namespacedName types.NamespacedName
+		fakeClient     ctrlruntimeclient.Client
+	}{
+		{
+			name:        "no pairs",
+			expected:    []string{},
+			expectedErr: false,
+			namespacedName: types.NamespacedName{
+				Namespace: "ci",
+				Name:      "ci-operator:latest",
+			},
+			fakeClient: fakectrlruntimeclient.NewClientBuilder().Build(),
+		},
+		{
+			name: "happy path",
+			expected: []string{
+				"registry.ci.openshift.org/ci/ci-operator@sha256:th15i5ah45h=quay.io/openshift/ci:ci_ci-operator_latest",
+				"registry.ci.openshift.org/ci/ci-operator@sha256:th15i5ah45h=quay.io/openshift/ci:20000101_sha256_th15i5ah45h",
+			},
+			expectedErr: false,
+			namespacedName: types.NamespacedName{
+				Namespace: "ci",
+				Name:      "ci-operator:latest",
+			},
+			fakeClient: fakectrlruntimeclient.NewClientBuilder().WithRuntimeObjects(
+				&imagev1.ImageStreamTag{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ci",
+						Name:      "ci-operator:latest",
+					},
+					Image: imagev1.Image{ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ci",
+						Name:      "sha256:th15i5ah45h",
+					},
+					}},
+			).Build(),
+		},
+		{
+			name:        "wrong imagestreamtag",
+			expected:    []string{},
+			expectedErr: true,
+			namespacedName: types.NamespacedName{
+				Namespace: "ci",
+				Name:      "ci-operator",
+			},
+			fakeClient: fakectrlruntimeclient.NewClientBuilder().Build(),
+		},
+		{
+			name:        "imagestreamtag not found",
+			expected:    []string{},
+			expectedErr: false,
+			namespacedName: types.NamespacedName{
+				Namespace: "ci",
+				Name:      "ci-operator:latest",
+			},
+			fakeClient: fakectrlruntimeclient.NewClientBuilder().WithRuntimeObjects(
+				&imagev1.ImageStreamTag{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ci",
+						Name:      "pj-rehearse:latest",
+					},
+					Image: imagev1.Image{ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ci",
+						Name:      "sha256:th15i5ad1ff3r3nth45h",
+					},
+					}},
+			).Build(),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			timeForTest := time.Date(2000, time.January, 1, 0, 0, 0, 0, time.UTC)
+			actual, err := createPairs(context.TODO(), tc.namespacedName, tc.fakeClient, timeForTest, logrus.NewEntry(logrus.StandardLogger()))
+			if err != nil && (err == nil || !tc.expectedErr) {
```

Comment:
The condition does not look correct to me. 
There is a util function from `testhelper.EquateErrorMessage` to compare the error msg.